.github/workflows/testing.yml

```diff
@@ -0,0 +1,81 @@
+name: macOS tests
+
+on:
+  - push
+  - pull_request
+
+
+jobs:
+  test_macos:
+
+    runs-on: macos-latest
+    strategy:
+      max-parallel: 3
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        numpy-version: ['--upgrade numpy']
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Setup openblas
+      run: |
+        # this setup is originally taken from the .travis.yml
+
+        # virtualenv needed for the multibuild steps
+        pip install virtualenv
+        brew install libmpc gcc@6 suitesparse
+        # The openblas binary used here was built using a gfortran older than 7,
+        # so it needs the older abi libgfortran.
+        export FC=gfortran-6
+        export CC=gcc-6
+        export CXX=g++-6
+        mkdir gcc_aliases
+        pushd gcc_aliases
+        ln -s `which gcc-6` gcc
+        ln -s `which g++-6` g++
+        ln -s `which gfortran-6` gfortran
+        # make gcc aliases in current dir
+        export PATH=$PWD/gcc_aliases:$PATH
+        popd
+        touch config.sh
+        git clone --depth=1 https://github.com/matthew-brett/multibuild.git
+
+        # designed for travis, but probably work on github actions
+        source multibuild/common_utils.sh
+        source multibuild/travis_steps.sh
+        before_install
+        export CFLAGS="-arch x86_64"
+        export CXXFLAGS="-arch x86_64"
+        printenv
+        wget https://3f23b170c54c2533c070-1c8a9b3114517dc5fe17b7c3f8c63a43.ssl.cf2.rackcdn.com/openblas-v0.3.7-macosx_10_9_x86_64-gf_1becaaa.tar.gz -O openblas.tar.gz
```

Comment:
We should avoid using the rackspace stuff moving forward since we'll get charged for that eventually; there's a whole other infrastructure for pulling in openblas from anaconda.org now--see `tools/openblas_support.py` in the SciPy repo.

I can clean that up after if you want though, since it is still in Travis too. We've migrated to this new way in Azure. I should probably add in the version checking too--we can check that the dynamic link was done to the right version of OpenBLAS.

Reply:
Amended.